dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/jfapchannel/impl/CommsOutboundChain.java

```diff
@@ -127,89 +159,130 @@ void unbindSecureFacet(OutboundSecureFacet oldFacet) {
         }
     }
 
-    private Map<String, Object> getTcpOptions() {
+    public Map<String, Object> getTcpOptions() {
         if (isAnyTracingEnabled() && tc.isEntryEnabled()) Tr.entry(this, tc, "getTcpOptions");
         Map<String, Object> tcpProps = tcpOptions.getConfiguration();
         if (isAnyTracingEnabled() && tc.isEntryEnabled()) exit(this, tc, "getTcpOptions", tcpProps);
         return tcpProps;
     }
 
+    @Reference(name = "nettyTlsProvider", cardinality = OPTIONAL, policyOption = GREEDY, unbind = "unbindTlsProviderService")
```

Comment:
This is not dynamic, so the reference belongs in the constructor parameter list.
Bind/unbind are unnecessary complications and extra code.
Also, we should use metatype to ensure we wait for this when it is expected.

Reply:
I also believe in part it should be in the constructor, originally tried it this way but ran into issues because the TLS bundles is loaded with the SSL feature used for creating Netty SSL configurations. So if a server is configured with just unencrypted traffic, the server will never call the constructor parameter list for the bundle and nothing will ever load. Maybe we can figure something out that'd be better for this, I'll DM you.